### PR TITLE
Popover for Cartesian charts v9

### PR DESCRIPTION
**Pull request by srmukher**
Popover for Cartesian Charts in v9.
Created a separate component for popover and moved the popover related logic outside of Cartesian base class.
Made changes for Cartesian and Line chart.

### CODE REVIEW

File: packages/react-components/react-charts-preview/src/components/CommonComponents/CartesianChart.tsx [hunk 1]
```diff
@@ -20,16 +17,15 @@ import {
...
-  pointTypes,
   calculateLongestLabelWidth,
   createYAxisLabels,
   ChartTypes,
   wrapContent,
   isRtl,
 } from '../../utilities/index';
-import { LegendShape, Shape } from '../Legends/index';
 import { SVGTooltipText } from '../../utilities/SVGTooltipText';
+import PopoverComponent from './PopoverComponent';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/CommonComponents/CartesianChart.tsx:27]: call it just Popover

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx [hunk 2]
```diff
@@ -966,7 +997,7 @@ export const LineChart: React.FunctionComponent<ILineChartProps> = React.forward
           .attr('y2', `${lineHeight - _yAxisScale(pointToHighlight.y)}`);
 
         setNearestCircleToHighlight(pointToHighlight);
-        setIsCalloutVisible(true);
+        updatePosition(mouseEvent.clientX, mouseEvent.clientY);
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:1000]: dont need to set isPopverOpen here?

File: packages/react-components/react-charts-preview/src/components/CommonComponents/PopoverComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      width: 0,
+      height: 0,
+    }),
+  };
+
+  const classes = usePopoverStyles_unstable(props);
+
+  return (
+    <Popover positioning={{ target: virtualElement }} open={props.isPopoverOpen} inline>
+      <PopoverSurface tabIndex={-1}>
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/CommonComponents/PopoverComponent.tsx:34]: need to add focusable group to keep the popover tabbable

File: packages/react-components/react-charts-preview/src/components/CommonComponents/PopoverComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        {/** single x point its corresponding y point of single line/bar in the chart will render in callout */}
+        {!props.customizedCallout && !props.isCalloutForStack && (
+          <ChartHoverCard
+            Legend={props.xCalloutValue ? props.xCalloutValue : props.legend}
+            YValue={props.yCalloutValue ? props.yCalloutValue : props.YValue}
+            color={props.color}
+            culture={props.culture}
+          />
+        )}
+      </PopoverSurface>
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/CommonComponents/PopoverComponent.tsx:48]: ToDo - Ask Abhishek to write tests for popover control

**Review by AtishayMsft — APPROVED** [review]

File: packages/react-components/react-charts-preview/src/components/CommonComponents/CartesianChart.tsx (hunk 1 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/CommonComponents/CartesianChart.tsx:27]: added in next PR

File: packages/react-components/react-charts-preview/src/components/CommonComponents/PopoverComponent.tsx (hunk 3 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/CommonComponents/PopoverComponent.tsx:34]: added in next PR

File: packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx (hunk 2 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/LineChart/LineChart.tsx:1000]: yes isPopoverOpen is set within updatePosition

**Pull request merged**